src/datepicker/datepicker-input.ts

```diff
@@ -62,6 +64,13 @@ export class NgbInputDatepicker implements OnChanges,
   private _disabled = false;
   private _model: NgbDate;
   private _zoneSubscription: any;
+  private _autoCloseSubscriber: Subscriber;
+
+  /**
+   * The element used to toggle the visibility of the datepicker.
+   * Used mainly to avoid handling automatically the datepicker visibility when clicking on this toggling element.
+   */
+  @Input('ngbDatepicker') togglingElement: HTMLElement | null;
```

Comment:
This was a breaking change for the in #2440 
I should check why it doesn't look like a breaking change here...

Reply:
Well, if there is a toggling element but it is not passed here, the feature will not work properly when clicking on the toggling element. That's a breaking change.